<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="rotl">
<title>Connecting data to Open Tree trees • rotl</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Connecting data to Open Tree trees">
<meta property="og:description" content="rotl">
<meta property="og:image" content="https://docs.ropensci.org/rotl/logo.svg">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">rotl</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">3.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="../articles/rotl.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/data_mashups.html">Connecting data to Open Tree trees</a>
    <a class="dropdown-item" href="../articles/meta-analysis.html">Using the Open Tree synthesis in a comparative analysis</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/rotl/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Connecting data to Open Tree trees</h1>
                        <h4 data-toc-skip class="author">David Winter</h4>
            
            <h4 data-toc-skip class="date">2023-06-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/rotl/blob/HEAD/vignettes/data_mashups.Rmd" class="external-link"><code>vignettes/data_mashups.Rmd</code></a></small>
      <div class="d-none name"><code>data_mashups.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="combining-data-from-otol-and-other-sources-">Combining data from OToL and other sources.<a class="anchor" aria-label="anchor" href="#combining-data-from-otol-and-other-sources-"></a>
</h2>
<p>One of the major goals of <code>rotl</code> is to help users combine data from other sources with the phylogenetic trees in the Open Tree database. This examples document describes some of the ways in which a user might connect data to trees from Open Tree.</p>
</div>
<div class="section level2">
<h2 id="get-open-tree-ids-to-match-your-data-">Get Open Tree IDs to match your data.<a class="anchor" aria-label="anchor" href="#get-open-tree-ids-to-match-your-data-"></a>
</h2>
<p>Let’s say you have a dataset where each row represents a measurement taken from one species, and your goal is to put these measurements in some phylogenetic context. Here’s a small example: the best estimate of the mutation rate for a set of unicellular Eukaryotes along with some other property of those species which might explain the mutation rate:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">csv_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"protist_mutation_rates.csv"</span>, package <span class="op">=</span> <span class="st">"rotl"</span><span class="op">)</span></span>
<span><span class="va">mu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">csv_path</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">mu</span></span></code></pre></div>
<pre><code><span><span class="co">##                     species       mu pop.size genome.size</span></span>
<span><span class="co">## 1   Tetrahymena thermophila 7.61e-12 1.12e+08    1.04e+08</span></span>
<span><span class="co">## 2    Paramecium tetraurelia 1.94e-11 1.24e+08    7.20e+07</span></span>
<span><span class="co">## 3 Chlamydomonas reinhardtii 2.08e-10 1.00e+08    1.12e+08</span></span>
<span><span class="co">## 4  Dictyostelium discoideum 2.90e-11 7.40e+06    3.40e+07</span></span>
<span><span class="co">## 5  Saccharomyces cerevisiae 3.30e-10 1.00e+08    1.25e+08</span></span>
<span><span class="co">## 6       Saccharomyces pombe 2.00e-10 1.00e+07    1.25e+08</span></span></code></pre>
<p>If we want to get a tree for these species we need to start by finding the unique ID for each of these species in the Open Tree database. We can use the Taxonomic Name Resolution Service (<code>tnrs</code>) functions to do this. Before we do that we should see if any of the taxonomic contexts, which can be used to narrow a search and avoid conflicts between different codes, apply to our group of species:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rotl/">rotl</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/tnrs_contexts.html">tnrs_contexts</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Possible contexts:</span></span>
<span><span class="co">##    Animals </span></span>
<span><span class="co">##       Birds, Tetrapods, Mammals, Amphibians, Vertebrates </span></span>
<span><span class="co">##       Arthropods, Molluscs, Nematodes, Platyhelminthes, Annelids </span></span>
<span><span class="co">##       Cnidarians, Arachnids, Insects </span></span>
<span><span class="co">##    Fungi </span></span>
<span><span class="co">##       Basidiomycetes, Ascomycetes </span></span>
<span><span class="co">##    All life </span></span>
<span><span class="co">##    Bacteria </span></span>
<span><span class="co">##       SAR group, Archaea, Excavata, Amoebozoa, Centrohelida </span></span>
<span><span class="co">##       Haptophyta, Apusozoa, Diatoms, Ciliates, Forams </span></span>
<span><span class="co">##    Land plants </span></span>
<span><span class="co">##       Hornworts, Mosses, Liverworts, Vascular plants, Club mosses </span></span>
<span><span class="co">##       Ferns, Seed plants, Flowering plants, Monocots, Eudicots </span></span>
<span><span class="co">##       Rosids, Asterids, Asterales, Asteraceae, Aster </span></span>
<span><span class="co">##       Symphyotrichum, Campanulaceae, Lobelia</span></span></code></pre>
<p>Hmm, none of those groups contain all of our species. In this case we can search using the <code>All life</code> context and the function <code>tnrs_match_names</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">taxon_search</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tnrs_match_names.html">tnrs_match_names</a></span><span class="op">(</span>names <span class="op">=</span> <span class="va">mu</span><span class="op">$</span><span class="va">species</span>, context_name <span class="op">=</span> <span class="st">"All life"</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">taxon_search</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="23%">
<col width="23%">
<col width="16%">
<col width="6%">
<col width="10%">
<col width="5%">
<col width="13%">
</colgroup>
<thead><tr class="header">
<th align="left">search_string</th>
<th align="left">unique_name</th>
<th align="left">approximate_match</th>
<th align="right">ott_id</th>
<th align="left">is_synonym</th>
<th align="left">flags</th>
<th align="right">number_matches</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">tetrahymena thermophila</td>
<td align="left">Tetrahymena thermophila</td>
<td align="left">FALSE</td>
<td align="right">180195</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">paramecium tetraurelia</td>
<td align="left">Paramecium tetraurelia</td>
<td align="left">FALSE</td>
<td align="right">568130</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">chlamydomonas reinhardtii</td>
<td align="left">Chlamydomonas reinhardtii</td>
<td align="left">FALSE</td>
<td align="right">33153</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">dictyostelium discoideum</td>
<td align="left">Dictyostelium discoideum</td>
<td align="left">FALSE</td>
<td align="right">160850</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">saccharomyces cerevisiae</td>
<td align="left">Saccharomyces cerevisiae</td>
<td align="left">FALSE</td>
<td align="right">356221</td>
<td align="left">FALSE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">saccharomyces pombe</td>
<td align="left">Schizosaccharomyces pombe</td>
<td align="left">FALSE</td>
<td align="right">990004</td>
<td align="left">TRUE</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<p>Good, all of our species are known to Open Tree. Note, though, that one of the names is a synonym. <em>Saccharomyces pombe</em> is older name for what is now called <em>Schizosaccharomyces pombe</em>. As the name suggests, the Taxonomic Name Resolution Service is designed to deal with these problems (and similar ones like misspellings), but it is always a good idea to check the results of <code>tnrs_match_names</code> closely to ensure the results are what you expect.</p>
<p>In this case we have a good ID for each of our species so we can move on. Before we do that, let’s ensure we can match up our original data to the Open Tree names and IDs by adding them to our <code>data.frame</code>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mu</span><span class="op">$</span><span class="va">ott_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/taxonomy-methods.html">unique_name</a></span><span class="op">(</span><span class="va">taxon_search</span><span class="op">)</span></span>
<span><span class="va">mu</span><span class="op">$</span><span class="va">ott_id</span> <span class="op">&lt;-</span> <span class="va">taxon_search</span><span class="op">$</span><span class="va">ott_id</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="find-a-tree-with-your-taxa">Find a tree with your taxa<a class="anchor" aria-label="anchor" href="#find-a-tree-with-your-taxa"></a>
</h2>
<p>Now let’s find a tree. There are two possible options here: we can search for published studies that include our taxa or we can use the ‘synthetic tree’ from Open Tree. We can try both approaches.</p>
<div class="section level3">
<h3 id="published-trees">Published trees<a class="anchor" aria-label="anchor" href="#published-trees"></a>
</h3>
<p>Before we can search for published studies or trees, we should check out the list of properties we can use to perform such searches:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/studies_properties.html">studies_properties</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $tree_properties</span></span>
<span><span class="co">##  [1] "ot:specifiedRoot"                 "ot:unrootedTree"                 </span></span>
<span><span class="co">##  [3] "ot:nodeLabelTimeUnit"             "xsi:type"                        </span></span>
<span><span class="co">##  [5] "ot:MRCAName"                      "ot:nodeLabelDescription"         </span></span>
<span><span class="co">##  [7] "ot:nearestTaxonMRCAOttId"         "ot:reasonsToExcludeFromSynthesis"</span></span>
<span><span class="co">##  [9] "ot:curatedType"                   "ot:nearestTaxonMRCAName"         </span></span>
<span><span class="co">## [11] "ot:studyId"                       "ot:nodeLabelMode"                </span></span>
<span><span class="co">## [13] "rootedge"                         "nodeById"                        </span></span>
<span><span class="co">## [15] "tb:type.tree"                     "ot:branchLengthDescription"      </span></span>
<span><span class="co">## [17] "ot:messages"                      "ot:rootNodeId"                   </span></span>
<span><span class="co">## [19] "ot:inGroupClade"                  "ot:ottId"                        </span></span>
<span><span class="co">## [21] "ot:branchLengthMode"              "ot:MRCAOttId"                    </span></span>
<span><span class="co">## [23] "ot:ottTaxonName"                  "tb:quality.tree"                 </span></span>
<span><span class="co">## [25] "tb:kind.tree"                     "ot:outGroupEdge"                 </span></span>
<span><span class="co">## [27] "ntips"                            "edgeBySourceId"                  </span></span>
<span><span class="co">## [29] "label"                            "meta"                            </span></span>
<span><span class="co">## [31] "tb:ntax.tree"                     "ot:tag"                          </span></span>
<span><span class="co">## [33] "ot:branchLengthTimeUnit"         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $study_properties</span></span>
<span><span class="co">##  [1] "tb:identifier.study.tb1"       "id"                           </span></span>
<span><span class="co">##  [3] "ot:taxonLinkPrefixes"          "ot:studyYear"                 </span></span>
<span><span class="co">##  [5] "prism:publicationName"         "dc:title"                     </span></span>
<span><span class="co">##  [7] "prism:creationDate"            "xmlns"                        </span></span>
<span><span class="co">##  [9] "skos:changeNote"               "skos:historyNote"             </span></span>
<span><span class="co">## [11] "ot:notIntendedForSynthesis"    "ot:otusElementOrder"          </span></span>
<span><span class="co">## [13] "dc:date"                       "xhtml:license"                </span></span>
<span><span class="co">## [15] "prism:startingPage"            "prism:publicationDate"        </span></span>
<span><span class="co">## [17] "ot:focalCladeOTTTaxonName"     "prism:section"                </span></span>
<span><span class="co">## [19] "prism:pageRange"               "ot:messages"                  </span></span>
<span><span class="co">## [21] "ot:studyPublicationReference"  "ot:curatorName"               </span></span>
<span><span class="co">## [23] "ot:dataDeposit"                "prism:number"                 </span></span>
<span><span class="co">## [25] "ot:agents"                     "dc:publisher"                 </span></span>
<span><span class="co">## [27] "treesById"                     "dcterms:bibliographicCitation"</span></span>
<span><span class="co">## [29] "ot:treesElementOrder"          "prism:modificationDate"       </span></span>
<span><span class="co">## [31] "tb:identifier.study"           "about"                        </span></span>
<span><span class="co">## [33] "ntrees"                        "nexmljson"                    </span></span>
<span><span class="co">## [35] "ot:comment"                    "dc:contributor"               </span></span>
<span><span class="co">## [37] "dc:creator"                    "prism:volume"                 </span></span>
<span><span class="co">## [39] "tb:title.study"                "treebaseId"                   </span></span>
<span><span class="co">## [41] "prism:endingPage"              "generator"                    </span></span>
<span><span class="co">## [43] "version"                       "ot:studyId"                   </span></span>
<span><span class="co">## [45] "ot:candidateTreeForSynthesis"  "otusById"                     </span></span>
<span><span class="co">## [47] "prism:doi"                     "dc:subject"                   </span></span>
<span><span class="co">## [49] "ot:studyPublication"           "ot:focalClade"                </span></span>
<span><span class="co">## [51] "ot:annotationEvents"           "ot:tag"                       </span></span>
<span><span class="co">## [53] "nexml2json"</span></span></code></pre>
<p>We have <code>ottIds</code> for our taxa, so let’s use those IDs to search for trees that contain them. Starting with our first species <em>Tetrahymena thermophila</em> we can use <code>studies_find_trees</code> to do this search.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/studies_find_trees.html">studies_find_trees</a></span><span class="op">(</span>property <span class="op">=</span> <span class="st">"ot:ottId"</span>, value <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="../reference/taxonomy-methods.html">ott_id</a></span><span class="op">(</span><span class="va">taxon_search</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    study_ids n_trees                       tree_ids candidate study_year</span></span>
<span><span class="co">## 1    ot_1587       1                          tree1                 2015</span></span>
<span><span class="co">## 2    ot_1589       1                          tree1                 2015</span></span>
<span><span class="co">## 3    ot_2037       4 tree10, tree11, tree12, tree13                 2018</span></span>
<span><span class="co">## 4     ot_409       2                   tree1, tree2     tree2       2015</span></span>
<span><span class="co">## 5     ot_564       1                        Tr85317   Tr85317       2015</span></span>
<span><span class="co">## 6     ot_579       1                        Tr60046                 2013</span></span>
<span><span class="co">## 7     ot_700       1                          tree1     tree1       2016</span></span>
<span><span class="co">## 8      ot_73       1                          tree1     tree1       2013</span></span>
<span><span class="co">## 9     ot_766       1                        Tr85440                 2015</span></span>
<span><span class="co">## 10    ot_767       1                          tree1     tree1       2016</span></span>
<span><span class="co">## 11     ot_87       1                        Tr64119   Tr64119       2014</span></span>
<span><span class="co">##                                                                                                                                    title</span></span>
<span><span class="co">## 1  'Phylogenomic analyses reveal subclass Scuticociliatia as the sister group of subclass Hymenostomatia within class Oligohymenophorea'</span></span>
<span><span class="co">## 2  'Phylogenomic analyses reveal subclass Scuticociliatia as the sister group of subclass Hymenostomatia within class Oligohymenophorea'</span></span>
<span><span class="co">## 3                                                                                                                                       </span></span>
<span><span class="co">## 4                                                                         Tree of life reveals clock-like speciation and diversification</span></span>
<span><span class="co">## 5     'The alveolate translation initiation factor 4E family reveals a custom toolkit for translational control in core dinoflagellates'</span></span>
<span><span class="co">## 6                                       'Convergent evolution of heat-inducibility during subfunctionalization of the Hsp70 gene family'</span></span>
<span><span class="co">## 7                                                                                                       'A new view of the tree of life'</span></span>
<span><span class="co">## 8                                           Deep relationships of Rhizaria revealed by phylogenomics: A farewell to Haeckel’s Radiolaria</span></span>
<span><span class="co">## 9                                                                                 'Bacterial proteins pinpoint a single eukaryotic root'</span></span>
<span><span class="co">## 10                 'Untangling the early diversification of eukaryotes: a phylogenomic study of the evolutionary origins of Centrohelida</span></span>
<span><span class="co">## 11                        'Dinoflagellate phylogeny revisited: Using ribosomal proteins to resolve deep branching dinoflagellate clades'</span></span>
<span><span class="co">##                                        study_doi n_matched_trees</span></span>
<span><span class="co">## 1  http://dx.doi.org/10.1016/j.ympev.2015.05.007               1</span></span>
<span><span class="co">## 2  http://dx.doi.org/10.1016/j.ympev.2015.05.007               1</span></span>
<span><span class="co">## 3    http://dx.doi.org/10.1038/s41586-018-0708-8               3</span></span>
<span><span class="co">## 4        http://dx.doi.org/10.1093/molbev/msv037               1</span></span>
<span><span class="co">## 5    http://dx.doi.org/10.1186/s12862-015-0301-9               1</span></span>
<span><span class="co">## 6      http://dx.doi.org/10.1186/1471-2148-13-49               1</span></span>
<span><span class="co">## 7   http://dx.doi.org/10.1038/nmicrobiol.2016.48               1</span></span>
<span><span class="co">## 8  http://dx.doi.org/10.1016/j.ympev.2012.12.011               1</span></span>
<span><span class="co">## 9      http://dx.doi.org/10.1073/pnas.1420657112               1</span></span>
<span><span class="co">## 10      http://dx.doi.org/10.1098/rspb.2015.2802               1</span></span>
<span><span class="co">## 11 http://dx.doi.org/10.1016/j.ympev.2013.10.007               1</span></span>
<span><span class="co">##            match_tree_ids</span></span>
<span><span class="co">## 1                   tree1</span></span>
<span><span class="co">## 2                   tree1</span></span>
<span><span class="co">## 3  tree10, tree13, tree11</span></span>
<span><span class="co">## 4                   tree2</span></span>
<span><span class="co">## 5                 Tr85317</span></span>
<span><span class="co">## 6                 Tr60046</span></span>
<span><span class="co">## 7                   tree1</span></span>
<span><span class="co">## 8                   tree1</span></span>
<span><span class="co">## 9                 Tr85440</span></span>
<span><span class="co">## 10                  tree1</span></span>
<span><span class="co">## 11                Tr64119</span></span>
<span><span class="co">##  [ reached 'max' / getOption("max.print") -- omitted 16 rows ]</span></span></code></pre>
<p>Well… that’s not very promising. We can repeat that process for all of the IDs to see if the other species are better represented.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hits</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">mu</span><span class="op">$</span><span class="va">ott_id</span>, <span class="va">studies_find_trees</span>, property <span class="op">=</span> <span class="st">"ot:ottId"</span>, detailed <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">hits</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span><span class="op">[[</span><span class="st">"n_matched_trees"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  51  51 128  71  17  88</span></span></code></pre>
<p>OK, most of our species are not in any of the published trees available. You can help fix this sort of problem by <a href="https://tree.opentreeoflife.org/curator" class="external-link">making sure you submit your published trees to Open Tree</a>.</p>
</div>
<div class="section level3">
<h3 id="a-part-of-the-synthesis-tree">A part of the synthesis tree<a class="anchor" aria-label="anchor" href="#a-part-of-the-synthesis-tree"></a>
</h3>
<p>Thankfully, we can still use the complete Tree of Life made from the combined results of all of the published trees and taxonomies that go into Open Tree. The function <code>tol_induced_subtree</code> will fetch a tree relating a set of IDs.</p>
<p>Using the default arguments you can get a tree object into your R session:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ott_in_tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/taxonomy-methods.html">ott_id</a></span><span class="op">(</span><span class="va">taxon_search</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="../reference/is_in_tree.html">is_in_tree</a></span><span class="op">(</span><span class="fu"><a href="../reference/taxonomy-methods.html">ott_id</a></span><span class="op">(</span><span class="va">taxon_search</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">tr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tol_induced_subtree.html">tol_induced_subtree</a></span><span class="op">(</span>ott_ids <span class="op">=</span> <span class="va">ott_in_tree</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in collapse_singles(tr, show_progress): Dropping singleton nodes with</span></span>
<span><span class="co">## labels: mrcaott2ott276, mrcaott2ott142555, mrcaott2ott1551, mrcaott2ott7623,</span></span>
<span><span class="co">## Chloroplastida ott361838, Chlorophyta ott979501, mrcaott185ott42071,</span></span>
<span><span class="co">## mrcaott185ott1426, mrcaott1426ott1544, mrcaott1544ott8659, mrcaott1544ott15345,</span></span>
<span><span class="co">## mrcaott1544ott9282, mrcaott9389ott818260, mrcaott9389ott23557,</span></span>
<span><span class="co">## mrcaott23557ott527099, mrcaott148ott902, SAR ott5246039, Alveolata ott266751,</span></span>
<span><span class="co">## Ciliophora (phylum in subkingdom SAR) ott302424, Intramacronucleata</span></span>
<span><span class="co">## ott340382, mrcaott1546ott1671, Conthreep ott5248773, mrcaott1671ott16129,</span></span>
<span><span class="co">## Peniculia ott1002116, Paramecium (genus in subkingdom SAR) ott568126,</span></span>
<span><span class="co">## mrcaott11752ott13570, Hymenostomatia ott5257367, Tetrahymena (genus in</span></span>
<span><span class="co">## subkingdom SAR) ott47284, mrcaott295406ott523463, mrcaott295406ott523462,</span></span>
<span><span class="co">## Amoebozoa ott1064655, mrcaott3973ott15653, mrcaott3973ott26103,</span></span>
<span><span class="co">## mrcaott26103ott273110, mrcaott26103ott229626, Dictyostelia ott835575,</span></span>
<span><span class="co">## mrcaott26103ott59686, Dictyosteliales ott4008839, Dictyosteliaceae ott4008841,</span></span>
<span><span class="co">## Dictyostelium ott999665, Opisthokonta ott332573, Nucletmycea ott5246132,</span></span>
<span><span class="co">## Fungi ott352914, mrcaott109ott3465, mrcaott109ott67172, mrcaott109ott1423,</span></span>
<span><span class="co">## mrcaott109ott69881, mrcaott109ott55157, mrcaott109ott9352, h2007-1 ott5584405,</span></span>
<span><span class="co">## Dikarya ott656316, mrcaott235ott1053, Saccharomycetales ott4085960,</span></span>
<span><span class="co">## Saccharomycetaceae ott989994, Saccharomyces ott908546, Schizosaccharomycetes</span></span>
<span><span class="co">## ott921286, Schizosaccharomycetidae ott5670481, Schizosaccharomycetales</span></span>
<span><span class="co">## ott508517, Schizosaccharomycetaceae ott990009, Schizosaccharomyces ott990008</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tr</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="fig-vignettes-subtree-1.png" alt=""><p class="caption">plot of chunk subtree</p>
</div>
</div>
<div class="section level3">
<h3 id="connect-your-data-to-the-tips-of-your-tree">Connect your data to the tips of your tree<a class="anchor" aria-label="anchor" href="#connect-your-data-to-the-tips-of-your-tree"></a>
</h3>
<p>Now we have a tree for of our species, how can we use the tree and the data together?</p>
<p>The package <code>phylobase</code> provide an object class called <code>phylo4d</code>, which is designed to represent a phylogeny and data associated with its tips. In order to get our tree and data into one of these objects we have to make sure the labels in the tree and in our data match exactly. That’s not quite the case at the moment (tree labels have underscores and IDs appended):</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mu</span><span class="op">$</span><span class="va">ott_name</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $`Tetrahymena thermophila`</span></span>
<span><span class="co">## [1] "Tetrahymena thermophila"</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Dictyostelium_discoideum_ott160850"</span></span></code></pre>
<p><code>rotl</code> provides a convenience function <code>strip_ott_ids</code> to deal with these.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/strip_ott_ids.html">strip_ott_ids</a></span><span class="op">(</span><span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span>, remove_underscores <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">tr</span><span class="op">$</span><span class="va">tip.label</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">mu</span><span class="op">$</span><span class="va">ott_name</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE TRUE TRUE TRUE TRUE TRUE</span></span></code></pre>
<p>Ok, now the tips are together we can make a new dataset. The <code><a href="https://rdrr.io/pkg/phylobase/man/phylo4d-methods.html" class="external-link">phylo4d()</a></code> functions matches tip labels to the row names of a <code>data.frame</code>, so let’s make a new dataset that contains just the relevant data and has row names to match the tree</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/fmichonneau/phylobase" class="external-link">phylobase</a></span><span class="op">)</span></span>
<span><span class="va">mu_numeric</span> <span class="op">&lt;-</span> <span class="va">mu</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mu"</span>, <span class="st">"pop.size"</span>, <span class="st">"genome.size"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">mu_numeric</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">mu</span><span class="op">$</span><span class="va">ott_name</span></span>
<span><span class="va">tree_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phylobase/man/phylo4d-methods.html" class="external-link">phylo4d</a></span><span class="op">(</span><span class="va">tr</span>, <span class="va">mu_numeric</span><span class="op">)</span></span></code></pre></div>
<p>And now we can plot the data and the tree together</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tree_data</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="fig-vignettes-unnamed-chunk-1-1.png" alt=""><p class="caption">plot of chunk unnamed-chunk-1</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="find-external-data-associated-with-studies-trees-and-taxa-from-open-tree">Find external data associated with studies, trees and taxa from Open Tree<a class="anchor" aria-label="anchor" href="#find-external-data-associated-with-studies-trees-and-taxa-from-open-tree"></a>
</h2>
<p>In the above example we looked for a tree that related species in another dataset. Now we will go the other way, and try to find data associated with Open Tree records in other databases.</p>
<div class="section level3">
<h3 id="get-external-data-from-a-study">Get external data from a study<a class="anchor" aria-label="anchor" href="#get-external-data-from-a-study"></a>
</h3>
<p>Let’s imagine you were interested in extending or reproducing the results of a published study. If that study is included in Open Tree you can find it via <code>studies_find_studies</code> or <code>studies_find_trees</code> and retrieve the published trees with <code>get_study</code>. <code>rotl</code> will also help you find external. The function <code>study_external_IDs</code> retrieves the DOI for a given study, and uses that to gather some more data:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">extra_data</span> <span class="op">&lt;-</span> <span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="fu"><a href="../reference/study_external_IDs.html">study_external_IDs</a></span><span class="op">(</span><span class="st">"pg_1980"</span><span class="op">)</span>, silent <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">inherits</a></span><span class="op">(</span><span class="va">extra_data</span>, <span class="st">"try-error"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">extra_data</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## External data identifiers for study </span></span>
<span><span class="co">##  $doi:  10.1016/j.ympev.2006.04.016 </span></span>
<span><span class="co">##  $pubmed_id:  16762568 </span></span>
<span><span class="co">##  $nucleotide_ids: vector of 58 IDs</span></span>
<span><span class="co">##  $external_data_url http://purl.org/phylo/treebase/phylows/study/TB2:S1575</span></span></code></pre>
<p>Here the returned object contains an <code>external_data_url</code> (in this case a link to the study in Treebase), a pubmed ID for the paper and a vector IDs for the NCBI’s nucleotide database. The packages <code>treebase</code> and <code>rentrez</code> provide functions to make use of these IDs within R.</p>
<p>As an example, let’s use <code>rentrez</code> to download the first two DNA seqences and print them.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rentrez/" class="external-link">rentrez</a></span><span class="op">)</span></span>
<span><span class="va">seqs</span> <span class="op">&lt;-</span> <span class="kw"><a href="https://rdrr.io/r/base/try.html" class="external-link">try</a></span><span class="op">(</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"nucleotide"</span>, id <span class="op">=</span> <span class="va">extra_data</span><span class="op">$</span><span class="va">nucleotide_ids</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, rettype <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span>, silent <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">inherits</a></span><span class="op">(</span><span class="va">seqs</span>, <span class="st">"try-error"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"NCBI temporarily down."</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## &gt;AM181011.1 Plectroninia neocaledoniense partial 28S rRNA gene, specimen voucher G316300 (Queensland Museum)</span></span>
<span><span class="co">## GCTAGCAAGCGCGTCGGTGGTTCAGCCGGCTGGTCTCGTCGAGTTGTCGGTGTGCGGATCCGAACGGACC</span></span>
<span><span class="co">## GCGGCCGATGGCGTCGGCGGGCAAGCTGTGGTGCACTCTGTCGGCGTGCGCGTCAGCGTCGGTTTCGGCC</span></span>
<span><span class="co">## GGACGACGAGGCGCTCGGGGAAGGTAGCTGGACCGGTCTTCGGTGCAGTGTTATAGCCCTGGGCCGCTGG</span></span>
<span><span class="co">## GTTCGGCGTTTGGGACCGAGGAGAGAGATGATCGCTGCAGCGCCTGTCTCCCTCTCGAGGGGGGCTAGCC</span></span>
<span><span class="co">## AGCCGCTGTTTGGGTGGCGTCACTGGCGGAGGACTGCACGCAGTGCTTCGCCGGTGGTCGTGTCCAGGCG</span></span>
<span><span class="co">## GGCGGTGTGGGTATAGAGGCGCTTAGGACGCTGGCGTCCAAATGGCCGTGCGCGACCCGTCTTGAAACAC</span></span>
<span><span class="co">## GGACCAAGGAGTCTAGCATGTGCGCGAGTCTTAGGGTGTGGAAGCCCTCGGGCGCAATGAAAGTGAAGGG</span></span>
<span><span class="co">## CCGTCGTCTCTCGGGGCTGCGGTGTGAGGTGAGAGCCGTCGCCGTCGGGTGGCGGTGCATCATCGGCCGG</span></span>
<span><span class="co">## TCCATCCTGCTCTCAGGAGGATCTGCGCAAGAGCGTGTTTGCTGGGACCCGAAAGATGGTGAACTATGCC</span></span>
<span><span class="co">## TGAATAGGGTGAAGCCAGAGGAAACTCTGGTGGAGGCTCGTAGCGGTTCTGACGTGCAAATCGATCGTCA</span></span>
<span><span class="co">## AATTTGGGTATAGGGGCGAAAGACTAATCGAACCATCTAGTAGCTGGTTCCCTCCGAAGTTTCCCTCAGG</span></span>
<span><span class="co">## ATAGCTGGAACTCGTCTTGACACAGTTTTATCAGGTAAAGCGAATGATTAGAGGTCTTGGGGGTGAAACA</span></span>
<span><span class="co">## CCCTCAACCTATTCTCAAACTTTAAATAGGTAAGAAGCGCGACTTGCTCAATTGAAGTGGCGCGCAGTGA</span></span>
<span><span class="co">## ATGTGAGTTCCAAGTGGGCCATTTTTGGTAAGCAGAACTGGCGATGCGGGATGAACCGAACGCTCGGTTA</span></span>
<span><span class="co">## AGGTGCCCAAGTCGACGCTCATCAGACCCCAGAAAAGGTGTTGGTCGATATAGACAGCAGGACGGTGGCC</span></span>
<span><span class="co">## ATGGAAGTCGGAATCCGCTAAGGAGTGTGTAACAACTCACCTGCCGAATCAACTAGCCCTGAAAATGGAT</span></span>
<span><span class="co">## GGCGCTCAAGCGTCGCACCTATACCGAGCCGTCGTGGTAAATGCCAGGCCACGACGAGTAGGAGGGCGCG</span></span>
<span><span class="co">## GTGGTCGTGACGCAGCCCTTGGCGCGAGCCTGGGCGAAACGGCCTCCGGTGCAGATCTTGGTGGTAGTAG</span></span>
<span><span class="co">## CAAATATTCAAATGAGAGCTTTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTGGACATGG</span></span>
<span><span class="co">## GTTAGTCGATCCTAAGAGATAGGGAAGTTCCGTGTGAAAGTGCGCAATGCGCTTCTGTGCTGCGCGCCTC</span></span>
<span><span class="co">## CTATCGAAAGGGAATCGGGTTAATATTCCCGAACCGGAAGGCGGATATCTCTGGCTCTCGGGTCAGGAGC</span></span>
<span><span class="co">## GGCAACGCAAGCGTACTGCGAGACGTCGGCGGGGGCTCCGGGAAGAGTTGTCTTTTCTTTTTAACGCAGT</span></span>
<span><span class="co">## CGCCATCCCTGGAATCGGTTTGCCCGGAGATAGGGTTGGCTGGCTCGGTAAAGCAGCACACTTCATGTGC</span></span>
<span><span class="co">## TGTCCGGTGCGCTCTCGACGGCCCTTGAAAATCGCAGGTGTGCATCGATTCTCGCATCCGGTCGTACTCA</span></span>
<span><span class="co">## TAACCGCATCAGGTCTCCAAGGT</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## &gt;AM181010.1 Eilhardia schulzei partial 28S rRNA gene, specimen voucher G316071 (Queensland Museum)</span></span>
<span><span class="co">## GCTAGTAATGTACGTTGGTGGTTCAGCCGGCTAGTCTTGTCGAGTCGTCGTGTGGTGGATCCGACTGGAC</span></span>
<span><span class="co">## CGTCCGCGGTGGTGTCGGCGGGCGAGCTGTGGTGCACTCTACGGACGTGCGCGTCAGCGTCGGTTCTCGA</span></span>
<span><span class="co">## TGGGCGATAAGGTGCGTGGGGGAAGGTGGCTCGGTCCTTGGGAACTGAGTGTTACAGACCCTGGTGCTGG</span></span>
<span><span class="co">## GCTCGTCGTGGGACCGAGGAGAGAGAGAGATGATCGCTGCGGCACCTGCCCCGTTGTCATTTTTCGGGGC</span></span>
<span><span class="co">## TAGCCAGCCGTTTGTCAGGTGTGCGTCGGACGTTGAGGACTGCACGCAGTGCTGGACGTGGAGGCGTGAT</span></span>
<span><span class="co">## CTGATGGCGGTGTGGGCATTAGAGGTGCCTAGGACGCTGGCGTCCAAATGGCCGTGCGCGACCCGTCTTG</span></span>
<span><span class="co">## AAACACGGACCAAGGAGTCTAACATGTGCGCGAGTCTTAGGGTGTGCAAGCCCTCGGGCGCAATGAAAGT</span></span>
<span><span class="co">## GAAGGCTCGGCGGCGCTAGTCGAGCTGAGGTGAGAGCCGTGGCCGTTGCATGTGGCGGCGGCGGCGCATC</span></span>
<span><span class="co">## ATCGGCCGGTCCATCCTGCTCTCAGGGGGATCCGAGCAAGAGCGTATTTGTTGGGACCCGAAAGATGGTG</span></span>
<span><span class="co">## AACTATGCCTGAATAGGGTGAAGCCAGAGGAAACTCTGGTGGAGGCTCGTAGCGATTCTGACGTGCAAAT</span></span>
<span><span class="co">## CGATCGTCAAATTTGGGTATAGGGGCGAAAGACTAATCGAACCATCTAGTAGCTGGTTCCCTCCGAAGTT</span></span>
<span><span class="co">## TCCCTCAGGATAGCTGGAGCTCTTGGACACAGTTTTATCAGGTAAAGCGAATGATCAGAGGTCTTGGGGG</span></span>
<span><span class="co">## TGAAACACCCTCAACCTATTCTCAAACTTTAAATCGGTAAGAAGCGCGACTTGCTGAATTGAAGCCGCGC</span></span>
<span><span class="co">## GCAAGCAATGTGAGTTCCAAGTGGGCCATTTTTGGTAAGCAGAACTGGCGATGCGGGATGAACCGAACGC</span></span>
<span><span class="co">## TGGGTTAAGGTGCCAAAGTCGACGCTCATCAGACCCCAGAAAAGGTGTTGGTTGATATAGACAGCAGGAC</span></span>
<span><span class="co">## GATGGCCATGGAAGTCGGAATCCGCTAAGGAGTGTGTAACAACTCACCTGCCGAATCAACTAGCCCTGAA</span></span>
<span><span class="co">## AATGGATGGCGCTCAAGCGTCGCACCTATACCGGGCCGTCGTCGCAAATGCCAGGCGACGACGAGTAGGA</span></span>
<span><span class="co">## GGGCGCAGTGGTCGTCATGCAGCCCTTGGCGTGAGCCTGGGTCAAACGGCCTCTGGTGCAGATCTTGGTG</span></span>
<span><span class="co">## GTAGTAGCAAATATTCAAATGAGAGCTTTGAAGACCGAAGTGGAGAAAGGTTCCATGTGAACAGCAGTTG</span></span>
<span><span class="co">## GACATGGGTTAGTCGATCCTAAGTGATAGGGGAGCTCCGTATGAAAGTGCGCAATCGGCCCTGCTTGTGT</span></span>
<span><span class="co">## CGCCTTGCGCCACCTATCGAAAGGGAATCGGGTTAATATTCCCGAACCGGAAGGCGGATTTTCTCTGGCT</span></span>
<span><span class="co">## CTCGGGTCAGGAGCGGCAACGCTAGCGAACCGCGAGACGTCGGCGGGGGCTCCGGGAAGAGTTGTCTTTT</span></span>
<span><span class="co">## CTTTTTAACGCAGTCGCCATCCCTGGAATCGGTTTGCCCGGAGATAGGGTTGGCTGGCTCGGTAAAGCAG</span></span>
<span><span class="co">## CACACTTCATGTGCTGTCCGGTGCGCTCTCGACGGCCCTTGAAAATCGCGGCGAGTGTAGTCTGATTTTC</span></span>
<span><span class="co">## GCATCCGTTCGTACTCATAACCGCATCAGGTCTCCAAGGT</span></span></code></pre>
<p>You could further process these sequences in R with the function <code>read.dna</code> from <code>ape</code> or save them to disk by specifying a file name with <code>cat</code>.</p>
</div>
<div class="section level3">
<h3 id="find-a-ott-taxon-in-another-taxonomic-database">Find a OTT taxon in another taxonomic database<a class="anchor" aria-label="anchor" href="#find-a-ott-taxon-in-another-taxonomic-database"></a>
</h3>
<p>It is also possible map an Open Tree taxon to a record in another taxonomic database. For instance, if we wanted to search for data about one of the tips of the sub-tree we fetched in the example above we could do so using <code>taxon_external_IDs</code>:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Tt_ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/taxon_external_IDs.html">taxon_external_IDs</a></span><span class="op">(</span><span class="va">mu</span><span class="op">$</span><span class="va">ott_id</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">Tt_ids</span></span></code></pre></div>
<pre><code><span><span class="co">##   source       id</span></span>
<span><span class="co">## 1  silva AY102613</span></span>
<span><span class="co">## 2   ncbi     5888</span></span>
<span><span class="co">## 3   gbif  7415807</span></span></code></pre>
<p>A user could then use <code>rgbif</code> to find locality records using the gbif ID or <code>rentrez</code> to get genetic or bibliometric data about from the NCBI’s databases.</p>
</div>
</div>
<div class="section level2">
<h2 id="what-next">What next<a class="anchor" aria-label="anchor" href="#what-next"></a>
</h2>
<p>The demonstration gets you to the point of visualizing your data in a phylogenetic context. But there’s a lot more you do with this sort of data in R. For instance, you could use packages like <code>ape</code>, <code>caper</code>, <code>phytools</code> and <code>mcmcGLMM</code> to perform phylogenetic comparative analyses of your data. You could gather more data on your species using packages that connect to trait databases like <code>rfishbase</code>, <code>AntWeb</code> or <code>rnpn</code> which provides data from the US National Phenology Network. You could also use <code>rentrez</code> to find genetic data for each of your species, and use that data to generate branch lengths for the phylogeny.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
